### PR TITLE
Update MLflow and Databricks docs

### PR DESCRIPTION
**Pull request by harupy**
Depends on #13699. Updates the existing mlflow and databricks examples.

### CODE REVIEW

File: docs/docs/integrations/providers/databricks.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ Databricks MLflow makes it more convenient to develop LangChain applications on
 Databricks MLflow AI Gateway
```
Comment (dbczumar) [inline docs/docs/integrations/providers/databricks.md:28]: @harupy Can we also update this page to show simpler completions examples for external / foundation model endpoints, and also show examples for chat & embeddings?

Comment (harupy) [reply docs/docs/integrations/providers/databricks.md:28]: `docs/docs/integrations/llms/databricks.ipynb` contains examples for external / foundation model endpoints. Can we use them?

Comment (dbczumar) [reply docs/docs/integrations/providers/databricks.md:28]: This section is out of date: https://github.com/langchain-ai/langchain/blob/fc02bb88b1633cbd9f7290fd33aeef1091f29f63/docs/docs/integrations/providers/databricks.md#databricks-as-an-llm-provider

Comment (harupy) [reply docs/docs/integrations/providers/databricks.md:28]: Got it, I'll update them.

File: docs/docs/integrations/providers/databricks.md [hunk 2]
```diff
@@ -25,17 +24,61 @@ Databricks provides a fully managed and hosted version of MLflow integrated with
...
-See [MLflow AI Gateway](/docs/integrations/providers/mlflow_ai_gateway).
-
 Databricks as an LLM provider
 -----------------------------
 
 The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks) illustrates the method to wrap Databricks endpoints as LLMs in LangChain. It supports two types of endpoints: the serving endpoint, which is recommended for both production and development, and the cluster driver proxy app, which is recommended for interactive development. 
 
-Databricks endpoints support Dolly, but are also great for hosting models like MPT-7B or any other models from the Hugging Face ecosystem. Databricks endpoints can also be used with proprietary models like OpenAI to provide a governance layer for enterprises.
+
+Databricks External Models
```
Comment (harupy) [inline docs/docs/integrations/providers/databricks.md:33]: @dbczumar Added `Databricks External Models` and `Databricks Foundation Model APIs`.

File: docs/docs/integrations/providers/databricks.md [hunk 3]
```diff
@@ -25,17 +24,60 @@ Databricks provides a fully managed and hosted version of MLflow integrated with
...
+```python
+from langchain.llms import DatabricksEmbeddings
+
+
+embeddings = DatabricksEmbeddings(endpoint="databricks-bge-large-en")
+print(embeddings.embed_query("hello")[:3])
+# -> [0.051055908203125, 0.007221221923828125, 0.003879547119140625, ...]
+```
 
 Databricks Dolly
```
Comment (harupy) [inline docs/docs/integrations/providers/databricks.md:82]: Not sure if this is something we still want to highlight, but no harm to keep it.

Comment (dbczumar) [reply docs/docs/integrations/providers/databricks.md:82]: Let's remove

Comment (harupy) [reply docs/docs/integrations/providers/databricks.md]: removed

**Comment by harupy** [comment]
@Benwilson2 Could you also take a look?

File: docs/docs/integrations/llms/databricks.ipynb [hunk 4]
```diff
@@ -23,24 +23,118 @@
...
+    "  databricks secrets create-scope <scope>\n",
+    "  databricks secrets put-secret <scope> openai-api-key --string-value $OPENAI_API_KEY\n",
+    "  ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following code creates a new serving endpoint with OpenAI's GPT-4 for chat and generates a response using the endpoint."
```
Comment (BenWilson2) [inline docs/docs/integrations/llms/databricks.ipynb:59]:
```suggestion
    "The following code creates a new serving endpoint with OpenAI's GPT-4 model for chat and generates a response using the endpoint."
```

File: docs/docs/integrations/providers/databricks.md [hunk 5]
```diff
@@ -25,19 +24,58 @@ Databricks provides a fully managed and hosted version of MLflow integrated with
 
 Databricks MLflow makes it more convenient to develop LangChain applications on Databricks. For MLflow tracking, you don't need to set the tracking uri. For MLflow Model Serving, you can save LangChain Chains in the MLflow langchain flavor, and then register and serve the Chain with a few clicks on Databricks, with credentials securely managed by MLflow Model Serving.
 
-Databricks MLflow AI Gateway
-----------------------------
+Databricks External Models
+--------------------------
 
-See [MLflow AI Gateway](/docs/integrations/providers/mlflow_ai_gateway).
+[Databricks External Models](https://docs.databricks.com/generative-ai/external-models/index.html) is designed to streamline the usage and management of various large language model (LLM) providers, such as OpenAI and Anthropic, within an organization. It offers a high-level interface that simplifies the interaction with these services by providing a unified endpoint to handle specific LLM related requests. The following example creates an endpoint that serves OpenAI's GPT-4 and generates a chat response from it:
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:30]:
```suggestion
[Databricks External Models](https://docs.databricks.com/generative-ai/external-models/index.html) is a service that is designed to streamline the usage and management of various large language model (LLM) providers, such as OpenAI and Anthropic, within an organization. It offers a high-level interface that simplifies the interaction with these services by providing a unified endpoint to handle specific LLM related requests. The following example creates an endpoint that serves OpenAI's GPT-4 model and generates a chat response from it:
```

File: docs/docs/integrations/providers/databricks.md [hunk 6]
```diff
@@ -25,19 +24,58 @@ Databricks provides a fully managed and hosted version of MLflow integrated with
...
-Databricks’ Dolly is an instruction-following large language model trained on the Databricks machine learning platform that is licensed for commercial use. The model is available on Hugging Face Hub as databricks/dolly-v2-12b. See the notebook [Hugging Face Hub](/docs/integrations/llms/huggingface_hub) for instructions to access it through the Hugging Face Hub integration with LangChain. 
+embeddings = DatabricksEmbeddings(endpoint="databricks-bge-large-en")
+print(embeddings.embed_query("hello")[:3])
+# -> [0.051055908203125, 0.007221221923828125, 0.003879547119140625, ...]
+```
+
+Databricks as an LLM provider
+-----------------------------
+
+The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks) illustrates the method to serve a custom model registered by MLflow as a Databricks endpoint.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:80]:
```suggestion
The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks) demonstrates how to serve a custom model that has been registered by MLflow as a Databricks endpoint.
```

File: docs/docs/integrations/providers/mlflow.mdx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+mlflow deployments start-server --config-path /path/to/config.yaml
+```
+
+## Example provided by `MLflow`
+
+>The `mlflow.langchain` module provides an API for logging and loading `LangChain` models.
+> This module exports multivariate LangChain models in the langchain flavor and univariate LangChain
+> models in the pyfunc flavor.
+
+See the [API documentation and examples](https://www.mlflow.org/docs/latest/python_api/mlflow.langchain).
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/mlflow.mdx:54]:
```suggestion
See the [API documentation and examples](https://www.mlflow.org/docs/latest/python_api/mlflow.langchain) for more information.
```

**Review by BenWilson2 — APPROVED** [review]
Very minor nits; LGTM :)

**Review by dbczumar — APPROVED** [review]
LGTM!

File: docs/docs/integrations/llms/databricks.ipynb [hunk 8]
```diff
@@ -23,24 +23,118 @@
     "* Cluster driver proxy app, recommended for iteractive development."
    ]
   },
+  {
```
[the remark this answers is not in the document]
Comment (harupy) [reply docs/docs/integrations/llms/databricks.ipynb:26]: updated!

File: docs/docs/integrations/providers/databricks.md [hunk 9]
```diff
@@ -25,19 +24,58 @@ Databricks provides a fully managed and hosted version of MLflow integrated with
...
-Databricks’ Dolly is an instruction-following large language model trained on the Databricks machine learning platform that is licensed for commercial use. The model is available on Hugging Face Hub as databricks/dolly-v2-12b. See the notebook [Hugging Face Hub](/docs/integrations/llms/huggingface_hub) for instructions to access it through the Hugging Face Hub integration with LangChain. 
+embeddings = DatabricksEmbeddings(endpoint="databricks-bge-large-en")
+print(embeddings.embed_query("hello")[:3])
+# -> [0.051055908203125, 0.007221221923828125, 0.003879547119140625, ...]
+```
+
+Databricks as an LLM provider
+-----------------------------
+
+The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks#wrapping-a-serving-endpoint-custom-model) demonstrates how to serve a custom model that has been registered by MLflow as a Databricks endpoint.
```
Comment (harupy) [inline docs/docs/integrations/providers/databricks.md:80]: checking this link is valid in the preview now.

File: docs/docs/integrations/providers/databricks.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (harupy) [reply docs/docs/integrations/providers/databricks.md]: updated!

**Pull request merged**